askama_shared/src/generator.rs

```diff
@@ -1533,19 +1506,35 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         DisplayWrap::Unwrapped
     }
 
-    fn visit_target(&mut self, buf: &mut Buffer, target: &'a Target<'_>) {
-        match *target {
+    fn visit_target(&mut self, buf: &mut Buffer, var: &Target<'a>) {
```

Comment:
In general, you have a lot going on in this commit. It would be great to split it into smaller pieces. For example, renaming this variable should and getting rid of the explicit lifetime could go in a separate commit so that it doesn't clutter up commits with more meaty changes.

If possible, it would be good to extract `visit_target_initialized()` and `write_let_needs_let` in different commits. Maybe the commit messages could also explain more clearly what you're trying to achieve here, since it's not that clear to me.